### PR TITLE
Polish up Req.Request docs

### PR DESCRIPTION
**Pull request by whatyouhide**

### CODE REVIEW

File: lib/req/request.ex [hunk 1]
```diff
@@ -122,45 +122,51 @@ defmodule Req.Request do
 
   Nothing is actually executed until we run the pipeline with `Req.Request.run_request/1`.
 
-  ### Request steps
+  ### Request Steps
```
Comment (whatyouhide) [inline lib/req/request.ex:125]: Consistent title casing.

File: lib/req/request.ex [hunk 2]
```diff
@@ -575,14 +590,15 @@ defmodule Req.Request do
       iex> Req.Request.update_private(req, :b, 11, & &1 + 1).private
       %{a: 1, b: 11}
   """
-  @spec update_private(t(), key :: atom(), default :: term(), (atom() -> term())) :: t()
+  @spec update_private(t(), key :: atom(), default :: term(), (term() -> term())) :: t()
```
Comment (whatyouhide) [inline lib/req/request.ex:593]: This was an error I think.

File: lib/req/request.ex [hunk 3]
```diff
@@ -632,6 +653,7 @@ defmodule Req.Request do
         inspect: &IO.inspect/1
       )
   """
+  @spec append_request_steps(t(), keyword(fun())) :: t()
```
Comment (whatyouhide) [inline lib/req/request.ex:656]: I think we might want to make the `request_step` and friends *public* types, as they are documented in the module doc anyway. Thoughts?

File: lib/req/request.ex [hunk 4]
```diff
@@ -556,6 +570,7 @@ defmodule Req.Request do
   @doc """
   Gets the value for a specific private `key`.
   """
+  @spec get_private(t(), atom(), default) :: term() | default when default: var
```
Comment (wojtekmach) [inline lib/req/request.ex:573]: What is `var`?

**Review by wojtekmach — APPROVED** [review]

File: lib/req/request.ex (hunk 3 above)
Comment (wojtekmach) [reply lib/req/request.ex:656]: Yeah we can add the type and use it in specs. I’d hold off on speccing request.request_specs though, I want to have an option to add metadata around steps in the future if I need to.

Comment (wojtekmach) [reply lib/req/request.ex:656]: Ah nevermind!

**Comment by wojtekmach** [comment]
Thank you!

**Pull request merged**